### PR TITLE
Refactor AppBuilder to reduce code duplication. Add compression type …

### PR DESCRIPTION
**Pull request by MichaelLiminal**
…enum. Add support for variable limapp file names (ulimapp and limapp). Add support for compressing limapps or just copying them uncompressed.

### CODE REVIEW

File: PlatformViewer/Scripts/App/BundleLoader/Impl/BundleAsyncLoadOperation.cs [hunk 1]
```diff
@@ -69,16 +69,16 @@ protected set
         public BundleAsyncLoadOperation(Experience experience) : base(experience)
         {
             mExperience = experience ?? throw new ArgumentNullException("experience");
-            CoroutineService.Instance.StartCoroutine(DoLoad(experience.Bytes));
+            CoroutineService.Instance.StartCoroutine(DoLoad(experience.Bytes,experience.CompressionType));
```
Comment (jarisrener) [inline PlatformViewer/Scripts/App/BundleLoader/Impl/BundleAsyncLoadOperation.cs]: space after comma

File: SDK/Editor/Serialization/AppPacker.cs [hunk 2]
```diff
@@ -137,7 +138,19 @@ private void DoPack()
             using (var inStream = File.OpenRead(tmpFile))
             using (var outStream = File.Create(mOutputPath))
             {
-                SevenZipHelper.Compress(inStream, outStream);
+                switch (mPack.CompressionType)
+                {
+                    case ECompressionType.LMZA:
+
```
Comment (jarisrener) [inline SDK/Editor/Serialization/AppPacker.cs]: redundant whitespace.

Comment (MichaelLiminal) [reply SDK/Editor/Serialization/AppPacker.cs]: Done

File: PlatformViewer/Scripts/App/BundleLoader/Impl/BundleAsyncLoadOperation.cs (hunk 1 above)
Comment (MichaelLiminal) [reply PlatformViewer/Scripts/App/BundleLoader/Impl/BundleAsyncLoadOperation.cs]: Done

**Pull request merged**